src/components/page/blocks/BlocksColumn.jsx

```diff
@@ -0,0 +1,73 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import Block from './index.jsx';
+
+function BlocksColumn({
+  col,
+  blocksGroups,
+  defaultLayout,
+  blockShared,
+  blockRows,
+}) {
+  const getLayout = layout => {
+    if (layout) {
+      const blockRow = layout.row || defaultLayout.row;
+      const blockCol = layout.col || defaultLayout.col;
+
+      return { row: blockRow, col: blockCol };
+    }
+
+    return defaultLayout;
+  };
+
+  const isInLayout = (layout, row) => {
+    const blockLayout = getLayout(layout);
+
+    return blockLayout.col === col && blockLayout.row === row;
+  };
+
+  const getRowBlockGroups = row =>
```

Comment:
after moving the initial reducer of `blocksGroups` up to `PageContainer` this method can be squashed down to a single `.reduce()` that filters out the blocks for the current row/col and then adds it back to the `blocksGroups` for the given row/col if there are any.